src/eigen/preconditioners.jl

```diff
@@ -28,7 +28,7 @@ mutable struct PreconditionerTPA{T <: Real}
     basis::PlaneWaveBasis
     kpt::Kpoint
     kin::AbstractVector{T}  # kinetic energy of every G
-    mean_kin::Union{Nothing, Vector{T}}  # mean kinetic energy of every band
+    mean_kin::Union{Nothing, AbstractVector{T}}  # mean kinetic energy of every band
```

Comment:
This should probably be specialized, ie parametrize the type with TA and have kin::TA, mean_kin::Union{Nothing, TA}. But it doesn't matter much.

Reply:
Agree. I'd have this:

```suggestion
mutable struct PreconditionerTPA{T <: Real, Tkin <: AbstractVector{T}}
    basis::PlaneWaveBasis
    kpt::Kpoint
    kin::Tkin  # kinetic energy of every G
    mean_kin::Tkin  # mean kinetic energy of every band or arbitrary shift
end
```
and I would pre-set the `mean_kin` with a single elements set to the default shift of 1 or something like that.